rosidl_typesupport_microxrcedds_c/QUALITY.md

```diff
@@ -0,0 +1,160 @@
+
+### License [3.iii]
+
+The license for **rosidl_typesupport_microxrcedds_c** is Apache 2.0, and a summary can be found in each source file.
+A full copy of the license can be found [here](LICENSE).
+
+### Copyright Statements [3.iv]
+
+**rosidl_typesupport_microxrcedds_c** copyright holder provide a statement of copyright in each source file.
+
+## Testing [4]
+
+### Feature Testing [4.i]
+
+**rosidl_typesupport_microxrcedds_c** provides tests which simulate typical usage, and they are located in the [`test/c` directory](../test/c).
+New features are required to have tests before being added as stated in [CONTRIBUTING](CONTRIBUTING.md).
+Current results can be found here:
+
+[![CI rosidl_typesupport_microxrcedds](https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/actions/workflows/ci.yml/badge.svg)](https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/actions/workflows/ci.yml)
+
+### Coverage [4.iii]
+
+**rosidl_typesupport_microxrcedds_c** checks the coverage of every commit. Last coverage can be seen in [Codecov](https://codecov.io/gh/micro-ROS/rosidl_typesupport_microxrcedds).
+
+### Linters and Static Analysis [4.v]
+
+**rosidl_typesupport_microxrcedds_c** [code style](https://github.com/eProsima/cpp-style) is enforced using [*uncrustify*](https://github.com/uncrustify/uncrustify).
+Among the CI tests there are tests that ensures that every pull request is compliant with the code style.
+The latest CI results can be seen [here](https://github.com/micro-ROS/rosidl_typesupport_microxrcedds/actions/workflows/ci.yml).
+
+**rosidl_typesupport_microxrcedds_c** uses and passes all the standard linters and static analysis tools for a C++ package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters).
+
+Results of the latest linter tests can be found [here](https://github.com/micro-ROS/rmw-rosidl_typesupport_microxrcedds/actions/workflows/ci.yml?query=branch%3Afoxy).
+
+## Dependencies [5]
+
+### Direct Runtime Dependencies [5.iii]
+
+**rosidl_typesupport_microxrcedds_c**  has the following runtime dependencies, which are at or above **Quality Level 2**:
+* `eProsima Micro CDR`: [QUALITY DECLARATION](https://github.com/eProsima/Micro-CDR/blob/master/QUALITY.md)
+* `rosidl_runtime_c`: [QUALITY DECLARATION](https://github.com/ros2/rosidl/blob/master/rosidl_runtime_c/QUALITY_DECLARATION.md)
+* `rosidl_typesupport_interface`: [QUALITY DECLARATION](https://github.com/ros2/rosidl/blob/master/rosidl_typesupport_interface/QUALITY_DECLARATION.md)
+
+## Platform Support [6]
+
+**rosidl_typesupport_microxrcedds_c** supports the all the platforms that are intended for. Those included GNU/Linux and POSIX systems and all the major embedded RTOS such as Zephyr RTOS, FreeRTOS and Nuttx. Windows is not included as Tier 1 platform because it is irrelevant for the nature of this package.
```

Comment:
```suggestion
**rosidl_typesupport_microxrcedds_c** is intended for micro-ROS and therefore to run on embedded platforms, so the officially supported platforms are not the same as in ROS 2. These include GNU/Linux and POSIX systems and all the major embedded RTOS such as Zephyr RTOS, FreeRTOS, and NuttX. Windows is not included as a Tier 1 platform because it is not relevant for this package due to its nature.
```